### ISSUE TITLE
coverage.py 6.3 regression - SIGTERM hang during time.sleep()

### ISSUE THREAD

**Issue by ben-spiller** [coveragepy/coveragepy#1340]
**Describe the bug**
I have a Python program that's doing a long (infinite) time.sleep(1000), and has no multiprocessing/threading. With coverage 6.2 it works but with 6.3 sending SIGTERM results in a hang. 

**To Reproduce**
1. Python 3.10
1. coverage.py 6.3.0. 6.2.0 does not hang. 
1. A python program that just does time.sleep(1000)
1. Sending a SIGTERM (on RHEL 8.5) does not terminate it with coverage.py 6.3 but does with 6.2. 

**Expected behavior**
SIGTERM should still terminate the process - the presence of coverage.py shouldn't break that. 

**Additional context**
Similar symptoms to GH-1310 but no multiprocessing involved here

**Comment by nedbat** [comment]
@ben-spiller Can you re-test with this fix for #1310 ?
```
pip install git+https://github.com/nedbat/coveragepy.git@803a5494ef23#egg=coverage
```

**Comment by nedbat** [comment]
Feel free to re-open this if you get more information.
